gcp/api/server.py

```diff
@@ -397,12 +397,15 @@ def bug_to_response(bug, include_details=True):
 
 def _get_bugs(bug_ids, to_response=bug_to_response):
   """Get bugs from bug ids."""
-  bugs = ndb.get_multi([ndb.Key(osv.Bug, bug_id) for bug_id in bug_ids])
-  return [
-      to_response(bug)
-      for bug in bugs
-      if bug and bug.status == osv.BugStatus.PROCESSED
-  ]
+  bugs = ndb.get_multi_async([ndb.Key(osv.Bug, bug_id) for bug_id in bug_ids])
```

Comment:
FYI: [This](https://cloud.google.com/appengine/docs/legacy/standard/python/ndb/queries#map) is the functional mapping I was thinking about yesterday (there's also an async version).

It only works on queries so some logic would have to move around, and you'd have to get each bug by id individually instead of using `get_multi_async`, so I'm not sure if it's really worth it. But if you were still looking into memory usage it might possibly maybe use less, though I'm not really convinced.

Reply:
Hmm it would require building a huge query of OR statements to make use of the map, not sure if that makes sense to do.
There is also this `done_callback` function which can be used for each of the futures, which might reduce memory usage by a bit more I can look into as well.